llvm/tools/llvm-cas-object-format/llvm-cas-object-format.cpp

```diff
@@ -643,15 +643,37 @@ static void dumpGraph(jitlink::LinkGraph &G, SharedStream &OS, StringRef Desc) {
   OS.applyLocked([&](raw_ostream &OS) { OS << Desc << ":\n" << Data; });
 }
 
+/// LinkGraph creates an anonymous symbol that covers the full EH frame section
+/// block and no edge points to it. The EH frame splitter may end up creating an
+/// identical symbol (that has an edge pointed to it) which can result in
+/// non-deterministic behavior in \p FlatV1 schema ingestion, because the
+/// symbols cannot be ordered in a deterministic manner. \p
+/// removeRedundantEHFrameSymbol() removes this redundant symbol before we run
+/// the EH frame splitter.
+/// FIXME: Should this be taken care of by the EH frame splitter?
+static void removeRedundantEHFrameSymbol(jitlink::LinkGraph &G) {
+  auto *EHFrame = G.findSectionByName("__TEXT,__eh_frame");
+  if (!EHFrame)
+    return;
+  for (jitlink::Symbol *Sym : G.defined_symbols()) {
+    if (&Sym->getBlock().getSection() == EHFrame) {
+      G.removeDefinedSymbol(*Sym);
+      break;
+    }
+  }
+}
+
 static Error createSplitEHFramePasses(jitlink::LinkGraph &G) {
   if (G.getTargetTriple().getObjectFormat() == Triple::MachO &&
       G.getTargetTriple().getArch() == Triple::x86_64) {
+    removeRedundantEHFrameSymbol(G);
     if (auto E = jitlink::createEHFrameSplitterPass_MachO_x86_64()(G))
       return E;
     if (auto E = jitlink::createEHFrameEdgeFixerPass_MachO_x86_64()(G))
       return E;
   } else if (G.getTargetTriple().getObjectFormat() == Triple::ELF &&
              G.getTargetTriple().getArch() == Triple::x86_64) {
+    removeRedundantEHFrameSymbol(G);
```

Comment:
Calling for ELF, but it's not going to work I think.

Reply:
I'll drop it from ELF path for now.

Reply:
I avoid calling on ELF and added a FIXME to check if it is necessary for ELF.